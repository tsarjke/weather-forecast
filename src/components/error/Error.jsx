import React from 'react';
import PropTypes from 'prop-types';
import './error.scss';

const Error = ({ text }) => (
  <div className="error" data-testid="error">
    <div className="error__content">
      <h2 className="error__title">Error!</h2>
      <p className="error__text">{text}</p>
    </div>
  </div>
);

Error.propTypes = {
  text: PropTypes.string,
};

Error.defaultProps = {
  text: 'Something went wrong',
};

export default Error;
