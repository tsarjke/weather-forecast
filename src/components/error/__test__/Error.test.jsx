import { render, screen } from '@testing-library/react';
import Error from '../Error';

describe('Error', () => {
  test('should render this component with the props text', () => {
    render(<Error text="test error" />);

    const errorElem = screen.getByTestId('error');
    expect(errorElem).toMatchSnapshot();
  });
});
