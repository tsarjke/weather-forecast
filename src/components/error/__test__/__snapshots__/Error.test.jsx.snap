// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error should render this component with the props text 1`] = `
<div
  class="error"
  data-testid="error"
>
  <div
    class="error__content"
  >
    <h2
      class="error__title"
    >
      Error!
    </h2>
    <p
      class="error__text"
    >
      test error
    </p>
  </div>
</div>
`;
